src/include/mutex.h

```diff
@@ -74,6 +74,16 @@ struct __wt_rwlock {
 
 struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_spinlock {
 	volatile int lock;
+
+	/*
+	 * We track acquisitions and time spent waiting for some locks. For
+	 * performance reasons and to make it possible to write generic code
+	 * that tracks statistics for different locks, we store the offset
+	 * of the statistics fields to be updated during lock acquisition.
+	 */
+	int16_t stat_count;		/* acquisitions */
```

Comment:
It took me quite a while to figure out how this works, even with the comment. I spent a while thinking "why can these never overflow".. Would you mind renaming the fields to make it clear that they are offsets to the counter location?